### PR TITLE
Use the same tag file for student list in InstructorStudentList and InstructorSearch, student search result #3995

### LINKED ISSUE #3995: Use the same tag file for student list in InstructorStudentList and InstructorSearch, student search result
They look exactly the same.

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #3995 
With slight behavioral change in the InstructorSearch side:
1. If the course has no section, the section column will be hidden.
2. Fixed the buggy sorting button on the table header (the toggle sort parameters were all off by one before).

### CODE REVIEW

**Comment by wkurniawan07** [comment]
@JosephineKwa hope you don't mind!

File: src/main/java/teammates/ui/controller/InstructorSearchPageData.java [hunk 1]
```diff
@@ -247,18 +250,55 @@ private void setSearchStudentsTables(StudentSearchResultBundle studentSearchResu
...
-        List<StudentRow> rows = new ArrayList<StudentRow>();      
+    private List<StudentListSectionData> createStudentRows(String courseId, 
+                                                           StudentSearchResultBundle studentSearchResultBundle) {
+        List<StudentListSectionData> rows = new ArrayList<StudentListSectionData>();      
         List<StudentAttributes> studentsInCourse = filterStudentsByCourse(
                                                        courseId, studentSearchResultBundle);
-        
+        Map<String, List<String>> sectionNameToTeamNameMap = new HashMap<String, List<String>>();
+        Map<String, List<StudentAttributes>> teamNameToStudentsMap = new HashMap<String, List<StudentAttributes>>();
+        Map<String, String> emailPhotoUrlMapping = new HashMap<String, String>();
```
Comment (JosephineKwa) [inline src/main/java/teammates/ui/controller/InstructorSearchPageData.java]: To keep naming consistent, perhaps you could name it emailToPhotoUrlMap instead

File: src/main/java/teammates/ui/template/SearchStudentsTable.java [hunk 2]
```diff
@@ -8,18 +8,34 @@
...
-    private List<StudentRow> studentRows;
+    private List<StudentListSectionData> sections;
+    private boolean hasSection;
     
-    public SearchStudentsTable(String courseId, List<StudentRow> studentRows) {
+    public SearchStudentsTable(String courseId, List<StudentListSectionData> sections) {
         this.courseId = courseId;
-        this.studentRows = studentRows;
+        this.sections = sections;
+        if (sections.size() == 1) {
```
Comment (JosephineKwa) [inline src/main/java/teammates/ui/template/SearchStudentsTable.java]: Maybe check for sections.size() <= 1 instead?

File: src/main/java/teammates/ui/template/SearchStudentsTable.java [hunk 3]
```diff
@@ -8,18 +8,34 @@
...
+            StudentListSectionData section = sections.get(0);
+            if (section.getSectionName().equals("None")) {
+                this.hasSection = false;
+            } else {
+                this.hasSection = true;
+            }
```
Comment (JosephineKwa) [inline src/main/java/teammates/ui/template/SearchStudentsTable.java]: For the if else statement above, perhaps you could collapse it into just `if (!section.getSectionName().equals("None"))`

File: src/main/java/teammates/ui/controller/InstructorSearchPageData.java (hunk 1 above)
Comment (JosephineKwa) [inline src/main/java/teammates/ui/controller/InstructorSearchPageData.java]: To keep the naming consistent, perhaps you could name it `emailToPhotoUrlMap` instead

File: src/main/java/teammates/ui/template/SearchStudentsTable.java (hunk 3 above)
Comment (yj-soh) [inline src/main/java/teammates/ui/template/SearchStudentsTable.java]: Or `this.hasSection = !section.getSectionName().equals("None")`.

File: src/main/webapp/WEB-INF/tags/instructor/studentList.tag [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+<%@ attribute name="hasSection" required="true" %>
+<%@ attribute name="sections" type="java.util.Collection" required="true" %>
+<%@ attribute name="fromStudentListPage" %>
+<table class="table table-responsive table-striped table-bordered margin-0">
+    <c:choose>
+        <c:when test="${not empty sections}">
+            <thead class="background-color-medium-gray text-color-gray font-weight-normal">
+                <tr id="resultsHeader-${courseIndex}">
+                    <th>Photo</th>
+                    <th id="button_sortsection-${courseIndex}" class="button-sort-none<c:if test="${not hasSection}"> hidden</c:if>" onclick="toggleSort(this,2)">
```
Comment (JosephineKwa) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: @yj-soh would it be better if this was a ternary operator instead?

Comment (yj-soh) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: I personally believe it's best to use `<c:if>` if there's only one option and `${ ? : }` if there are two.

We don't have any standard regarding this, though, so both are fine.

Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: Same, ternary looks redundant if either option is an empty string.

Comment (JosephineKwa) [inline src/main/webapp/WEB-INF/tags/instructor/studentList.tag]: :+1:

**Comment by JosephineKwa** [comment]
Looks good to merge

**Pull request merged**